src/main/scala/cromwell/binding/formatter/SyntaxFormatter.scala

```diff
@@ -0,0 +1,122 @@
+import cromwell.binding.types.WdlType
+
+import scala.collection.JavaConverters._
+
+import cromwell.binding._
+import cromwell.parser.WdlParser.{Terminal, AstList, AstNode, Ast}
+
+trait SyntaxHighlighter {
+  def keyword(s: String): String = s
+  def name(s: String): String = s
+  def section(s: String): String = s
+  def wdlType(t: WdlType): String = t.toWdlString
+  def variable(s: String): String = s
+  def alias(s: String): String = s
+  def command(s: String): String = s
+  def function(s: String): String = s
+}
+
+object NullSyntaxHighlighter extends SyntaxHighlighter
+
+object TerminalSyntaxHighlighter extends SyntaxHighlighter {
+  override def keyword(s: String): String = s"\033[38;5;214m$s\033[0m"
+  override def name(s: String): String = s"\033[38;5;253m$s\033[0m"
+  override def section(s: String): String = s
+  override def wdlType(t: WdlType): String = s"\033[38;5;33m${t.toWdlString}\033[0m"
+  override def variable(s: String): String = s"\033[38;5;112m$s\033[0m"
+  override def alias(s: String): String = s
+  override def command(s: String): String = s //s"\033[48;5;243m$s\033[0m"
+  override def function(s: String): String = s"\033[38;5;13m$s\033[0m"
+}
+
+object SyntaxFormatter {
+  def highlighter = TerminalSyntaxHighlighter
+  val indent = 2
+  def format(binding: WdlBinding): String = {
+    val imports = binding.imports.map(formatImport) match {
+      case v if v.size > 0 => v.mkString("\n") + "\n\n"
+      case v => ""
+    }
+    val definitions = for(node <- binding.ast.getAttribute("definitions").asInstanceOf[AstList].asScala.toVector) yield {
+      node match {
+        case a:Ast if a.getName == "Workflow" => formatWorkflow(binding.workflows.head)
+        case a:Ast if a.getName == "Task" => formatTask(binding.findTask(text(a.getAttribute("name"))).getOrElse {
+          throw new UnsupportedOperationException("Shouldn't happen")
+        })
+      }
+    }
+    s"$imports${definitions.mkString("\n\n")}"
+  }
+  private def formatImport(imp: Import): String = {
+    val namespace = imp.namespace match {
+      case Some(ns) => s" as $ns"
+      case None => ""
+    }
+    s"${highlighter.keyword("import")} '${imp.uri}'$namespace"
+  }
+  private def formatTask(task: Task): String = {
+    val outputs = if(task.outputs.size > 0) formatOutputs(task.outputs, 1) else ""
+    val command = formatCommandSection(task.command, 1)
+    val sections = List(command, outputs)
+    val header = s"""${highlighter.keyword("task")} ${highlighter.name(task.name)} {
+       |${sections.mkString("\n")}
+       |}"""
+     .stripMargin
+    header
+  }
+  private def formatCommandSection(command: Command, level:Int): String = {
+    val section = s"""${highlighter.section("command")} {
+        |${formatCommand(command, level+1)}
+        |}"""
+    indentText(section.stripMargin, level)
+  }
+  private def formatCommand(command: Command, level:Int) = {
+    val abstractCommand = command.parts.map {
```

Comment:
This is a general thought (so attn @mcovarr) - we're starting to accumulate these multi-case higher order functions throughout the codebase. It's been bugging me as it's not a construct I've seen often in my scala-y travels which has made me think there's a more idiomatic way to handle these situations.

I wonder if things would read better with multiple chained HOFs (in this case, `map`). There's an obvious performance penalty but a) Most of this stuff it wouldn't matter anyways and/or b) conversion to `stream` (and I think `iterator`) would lead to the compiler optimizing a single pass anyways